modules/targetAcquisition/taxi/Taxi.py

```diff
@@ -0,0 +1,331 @@
+                print(f"Calculated distance: {self.distanceFromBox}")
+                self.state = "TRACK"
+                print("Set variable self.state to TRACK")
+        else:
+            print("Error: invalid state selected")
+
+    def main(self, cap):
+        """
+        Main operations: getting camera input and passing the image to appropriate methods
+
+        Parameters
+        ----------
+        cap : np.ndarray
+            the current captured rgb image from the camera
+        """
+
+        self.frame = cap
+        if self.state == "BOX":
+            self.bbox = self.yolo.detect_boxes(self.frame)
+            for (topLeft, botRight) in self.bbox:
+                cv2.rectangle(self.frame, topLeft,
+                                botRight, (0, 0, 255), 2)
+
+            # If initial run, expect 5 boxes
+            # If recalibrating, expect at least 1 box (less accurate but there's no other way)
+            if not self.recalibrate:
+                expCount = self.expectedCount
+            else:
+                print("Recalibrating in BOX mode. Expecting 1 box")
+                expCount = 1
+
+            # YOLO can't move on until all 5 boxes stay in frame consistently/continuously for more than frameCount frames
+            if len(self.bbox) >= expCount:
+                self.frameCount += 1
+            else:
+                frameCount = 0
+
+            if frameCount == self.numStableFrames:
+                print("Got the required number of stable frames")
+                self.set_state("TRACK")
+                frameCount = 0
+
+            self.totalWait += 1
+            if self.totalWait > 250:
+                print(f"Waited {self.totalWait} frames without finding the expected number of bbox. Switching to human control.")
+
+        # Switch to track when all 5 boxes are in view
+        if self.state == "TRACK":
+            # print("In tracking mode")
+            found, bbox = self.tracker.update(self.frame)
+
+            # Tracking succeeds and plane isn't already close enough
+            # print(f"Found: {found}, distance from box: {self.distanceFromBox}, min distance from box: {self.minDistanceFromBox}")
+            if found and self.distanceFromBox > self.minDistanceFromBox:
+                # print("Tracking succeeds and plane isn't already close enough")
+                p1 = (int(bbox[0]), int(bbox[1]))
+                p2 = (int(bbox[0] + bbox[2]), int(bbox[1] + bbox[3]))
+                self.bbox = [(p1, p2)]
+                cv2.rectangle(self.frame, p1, p2, (255, 0, 0), 2, 1)
+
+                # While not at half the distance from the box, keep moving forward
+                # if distanceCount < self.distanceFromBox // 2:
+                if self.distanceCount < 500:
+                    # print("While not at half the distance from the box, keep moving forward")
+                    # Issue the next movement command every couple of frames
```

Comment:
Again we never actually end up sending a command anywhere in this file. This needs to be fixed